### ISSUE TITLE
Support batch normalization

### ISSUE THREAD

**Issue by daylen** [szagoruyko/loadcaffe#63]
Batch Normalization was added to Caffe in October 2015. (See https://github.com/BVLC/caffe/pull/3229 and https://github.com/BVLC/caffe/pull/3299). The caffe.proto file in this repo is from April 2015, so it doesn't know about Batch Normalization and can't load newer model definition files.

**Comment by szagoruyko** [comment]
thanks, are there models in caffe format with batchnorm available? should be quick to add, just need a test. resnets won't work.

**Comment by joshpowell** [comment]
@szagoruyko You're saying that you cannot use resnets for this test, or you're saying resnets won't work till you get this test working?

**Comment by szagoruyko** [comment]
@joshpowell resnets have non-sequential structure which loadcaffe doesn't support now

**Comment by nhynes** [comment]
<shamelessplug>
@joshpowell you might want to give [caffegraph](https://github.com/nhynes/caffegraph) a try. It's not yet as fully featured as loadcaffe (it's missing a few modules), but it will load your resnet just fine.
</shamelessplug>

**Comment by wuhuo714** [comment]
Hi, could loadcaffe load models in caffe format with batch normalization,now?

**Comment by szagoruyko** [comment]
@wuhuo714 no

**Comment by wuhuo714** [comment]
@szagoruyko Will batchnorm be available soon?

**Comment by szagoruyko** [comment]
@wuhuo714 don't have time unfortunately

**Comment by wuhuo714** [comment]
@szagoruyko Thanks a lot,anywhere!

**Comment by wuhuo714** [comment]
Hi,if my caffemodel has batchnorm layers which loadcaffe can not recognize, can the following layers like Relu get its correponding weights?It has confused me a lot.

**Comment by wuhuo714** [comment]
Hi,@szagoruyko,if my caffemodel has batchnorm layers which loadcaffe can not recognize, can the following layers like Relu get its correponding weights?It has confused me a lot.

**Comment by szagoruyko** [comment]
@wuhuo714 no, it will ignore layers it can't recognize

**Comment by wuhuo714** [comment]
Although the layers it can't recognize are ignored, will the weights of these unrecognizable layers be used in the following layers it recoginze? 
  Is that the reason why the following layers like conv can not get their corresponding weights?

**Comment by szagoruyko** [comment]
@wuhuo714 weights of unrecognized layers will not be loaded at all

**Comment by wuhuo714** [comment]
OK,it's very kind of you to answer my doubts!

**Comment by wuhuo714** [comment]
@szagoruyko ok, it's very kind of you to answer my doubts!

**Comment by Fourure** [comment]
Hi @szagoruyko,
First thank you for your work, your package is very useful for me (and probably for lot of peoples). 
I'm also interested by loading a caffe model with batch normalization. Is it still not available ?
